packages/workspace/src/browser/workspace-duplicate-handler.ts

```diff
@@ -0,0 +1,63 @@
+/********************************************************************************
+ * Copyright (C) 2018 Ericsson and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+import URI from '@theia/core/lib/common/uri';
+import { injectable, inject } from 'inversify';
+import { WorkspaceService } from './workspace-service';
+import { FileSystem } from '@theia/filesystem/lib/common/filesystem';
+import { UriCommandHandler } from '@theia/core/lib/common/uri-command-handler';
+import { FileSystemUtils } from '@theia/filesystem/lib/common/filesystem-utils';
+
+@injectable()
+export class WorkspaceDuplicateHandler implements UriCommandHandler<URI[]> {
```

Comment:
just curious: why **Workspace**DuplicateHandler ? it does not duplicate **Workspace**, does it ? :)

Reply:
ok talked with Vincent in person: there is a WorkspaceDeleteHandler, and here we are just following the "naming convention"